### ISSUE TITLE
Issue with TableView inside TabMan

### ISSUE THREAD

**Issue by markuplab** [uias/Tabman#485]
### New Issue Checklist

- [ +] Updated Tabman to the latest version
- [ +] I took a look at the [Troubleshooting Guide](https://github.com/uias/Tabman/blob/master/Docs/Troubleshooting.md)
- [+ ] I searched for [existing GitHub issues](https://github.com/uias/Tabman/issues)

### Issue Description
Hello. I have next VC:

```
class BSTabsViewController: TabmanViewController {
  typealias Tabs = TMBarView<
    TMConstrainedHorizontalBarLayout,
    BSLabelBarButton,
    TMLineBarIndicator
  >
  
  var viewControllers: [TabsItemData] = []
  
  override func viewDidLoad() {
    super.viewDidLoad()
    
    self.dataSource = self
    
    // Create bar
    let bar = Tabs()
    
    bar.buttons.customize { button in
      button.font = .nerisBlack(ofSize: 16)
      button.tintColor = .bs_white
      button.selectedTintColor = .bs_white
    }
    
    bar.indicator.weight = .medium
    bar.indicator.tintColor = .bs_skyBlue
    
    let systemBar = bar.systemBar()
    
    systemBar.backgroundStyle = .flat(color: .bs_blue)
    
    // Add to view
    addBar(
      systemBar,
      dataSource: self,
      at: .top
    )
    
    let border = systemBar.createViewBackedBorder(side: .bottom, thickness: 2, color: .bs_black, bottomOffset: -1)
    systemBar.addSubview(border)
  }
}

extension BSTabsViewController: PageboyViewControllerDataSource, TMBarDataSource {
  func barItem(for bar: TMBar, at index: Int) -> TMBarItemable {
    let tabData: TabsItemData = self.viewControllers[index]
    
    return TMBarItem(
      title: tabData.title
    )
  }
  
  func numberOfViewControllers(in pageboyViewController: PageboyViewController) -> Int {
    return viewControllers.count
  }
  
  func viewController(for pageboyViewController: PageboyViewController,
                      at index: PageboyViewController.PageIndex) -> UIViewController? {
    return viewControllers[index].viewController
  }
  
  func defaultPage(for pageboyViewController: PageboyViewController) -> PageboyViewController.Page? {
    return nil
  }
}
```

Then i have class that extends:
```
import UIKit

class EventsTabsViewController: BSTabsViewController {
  let provider: ServicesProvider
  
  init (provider: ServicesProvider) {
    self.provider = provider
    super.init(nibName: nil, bundle: nil)
  }
  
  required init?(coder aDecoder: NSCoder) {
    fatalError("init(coder:) has not been implemented")
  }
  
  override func viewDidLoad() {
    let reactor = EventsViewControllerReactor(
      eventsService: self.provider.eventsService,
      gameDataService: self.provider.gameDataService
    )
    
    let upcomingVC = EventsViewController(reactor: reactor, type: .upcoming)
    let currentVC = EventsViewController(reactor: reactor, type: .current)
    
    self.viewControllers = [
      TabsItemData(title: "Current", viewController: currentVC),
      TabsItemData(title: "Upcoming", viewController: upcomingVC)
    ]
    
    super.viewDidLoad()
  }
}
```

```
final class EventsViewController: BaseViewController, View {
  //... Logic here

  // MARK: UI
  let tableView = UITableView().then {
    $0.backgroundColor = .clear
    $0.register(Reusable.eventCell)
  }
  // MARK: Layout
  override func viewDidLoad() {
    super.viewDidLoad()
    self.view.addSubview(self.tableView)
  }
  
  override func setupConstraints() {
    self.tableView.snp.makeConstraints { make in
      make.edges.equalToSuperview()
    }
  }
  
  // ... Logic here
}
```

And i got next error:
```
[TableView] Warning once only: UITableView was told to layout its visible cells and other contents without being in the view hierarchy (the table view or one of its superviews has not been added to a window). This may cause bugs by forcing views inside the table view to load and perform layout without accurate information (e.g. table view bounds, trait collection, layout margins, safe area insets, etc), and will also cause unnecessary performance overhead due to extra layout passes. Make a symbolic breakpoint at UITableViewAlertForLayoutOutsideViewHierarchy to catch this in the debugger and see what caused this to occur, so you can avoid this action altogether if possible, or defer it until the table view has been added to a window. Table view: <UITableView: 0x7fa352027800; frame = (0 0; 414 896); clipsToBounds = YES; gestureRecognizers = <NSArray: 0x600001e99110>; layer = <CALayer: 0x60000109da60>; contentOffset: {0, 0}; contentSize: {414, 0}; adjustedContentInset: {0, 0, 0, 0}; dataSource: <RxCocoa.RxTableViewDataSourceProxy: 0x6000034ef960>
```

I tested without TabMan, and there no any errors.

**Comment by sleekuser** [comment]
@markuplab The quick fix is not to set `dataSource` inside `viewDidLoad` method, but to implement something like,

```
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)

        if dataSource == nil {
            dataSource = self
        }
    }
```
I think the issue is happening because Tabman is calculating insets of Pageboy view controllers before they are added to the view hierarchy. `UITableView` doesn't seem to like `layoutIfNeeded` without having a superview for proper cell size calculations.

**Comment by msaps** [comment]
@sleekuser @markuplab this should have been resolved with #487 👍

**Comment by sleekuser** [comment]
@msaps You're awesome mate!!!
